File: src/main/kotlin/limia/Exception/EntityNotFoundException.java
package limia.Exception;

import kotlin.reflect.KClass;

import java.util.ArrayList;

/**
 * Created by workstation on 16/04/2017.
 */
public class EntityNotFoundException extends Exception {

    private ArrayList<KClass> types = new ArrayList<KClass>();

    public ArrayList<KClass>  getTypes() {
        return types;
    }

    public void setTypes(ArrayList<KClass>  types) {
        this.types = types;
    }

    public void addEntityType(KClass type) {
        types.add(type);
    }

    public EntityNotFoundException() {
    }

    public EntityNotFoundException(String message) {
        super(message);
    }

    public EntityNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }

    public EntityNotFoundException(Throwable cause) {
        super(cause);
    }

    public EntityNotFoundException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: src/main/kotlin/limia/Exception/EntityNotFoundException.kt
package limia.Exception

import kotlin.reflect.KClass

import java.util.ArrayList

/**
 * Created by workstation on 16/04/2017.
 */
class EntityNotFoundException : Exception {

    var types = ArrayList<KClass<*>>()

    fun addEntityType(type: KClass<*>) {
        types.add(type)
    }

    constructor()

    constructor(message: String) : super(message)

    constructor(message: String, cause: Throwable) : super(message, cause)

    constructor(cause: Throwable) : super(cause)

    constructor(message: String, cause: Throwable, enableSuppression: Boolean, writableStackTrace: Boolean)
            : super(message, cause, enableSuppression, writableStackTrace)
}
